x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/rule_assets/prebuilt_rule_assets_client/methods/fetch_assets_by_version.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract, SavedObjectsRawDocSource } from '@kbn/core/server';
+import { invariant } from '../../../../../../../../common/utils/invariant';
+import { MAX_PREBUILT_RULES_COUNT } from '../../../../../rule_management/logic/search/get_existing_prepackaged_rules';
+import type { PrebuiltRuleAsset } from '../../../../model/rule_assets/prebuilt_rule_asset';
+import { PREBUILT_RULE_ASSETS_SO_TYPE } from '../../prebuilt_rule_assets_type';
+import { validatePrebuiltRuleAssets } from '../../prebuilt_rule_assets_validation';
+import type { RuleVersionSpecifier } from '../../../rule_versions/rule_version_specifier';
+
+/**
+ * Fetches prebuilt rule assets for specified rule versions.
+ *
+ * Takes a list of objects with "rule_id" and "version" properties.
+ * Returns full prebuilt rule.
+ *
+ * @param savedObjectsClient - The saved objects client used to query the saved objects store
+ * @param versions - An array of rule version specifiers, each containing a rule_id and version.
+ * @returns A promise that resolves to an array of prebuilt rule assets.
+ */
+export async function fetchAssetsByVersion(
+  savedObjectsClient: SavedObjectsClientContract,
+  versions: RuleVersionSpecifier[]
+): Promise<PrebuiltRuleAsset[]> {
+  if (versions.length === 0) {
+    // NOTE: without early return it would build incorrect filter and fetch all existing saved objects
+    return [];
+  }
+
+  const soIds = versions.map(
+    (version) => `${PREBUILT_RULE_ASSETS_SO_TYPE}:${version.rule_id}_${version.version}`
+  );
+
+  const searchResult = await savedObjectsClient.search<
+    SavedObjectsRawDocSource & {
+      [PREBUILT_RULE_ASSETS_SO_TYPE]: PrebuiltRuleAsset;
+    }
+  >({
+    type: PREBUILT_RULE_ASSETS_SO_TYPE,
+    namespaces: ['default'],
```

Comment:
Have you checked it's safe to use the `default` namespace? Shall we use the current namespace instead?

My concern is that the current user may not have permissions to access the `default` namespace and the SO client may restrict SOs access in that case.